### PR TITLE
Using runZoned to remove test pollution

### LINKED ISSUE #53: yaml_edit/test/fuzz_test.dart is very noisy
This test prints a lot of:
```
line 1, column 1: Warning: unknown directive.                                          
  ╷                                                                                                                                                                           
1 │ %*.*s                                                                                                                                                                     
  │ ^^^^^                                                                                                                                                                     
  ╵    
```
It's probably best if we can cleanup this noise, so that it doesn't pollute our test logs.

/cc @walnutdust

### PR DESCRIPTION
**Pull request by walnutdust**
Closes #53

### CODE REVIEW

File: yaml_edit/test/naughty_test.dart [hunk 1]
```diff
@@ -21,11 +23,15 @@ void main() {
...
-      expect(value, isA<String>());
-      expect(value, equals(string));
+      runZoned(() {
+        expect(() => doc.update([], string), returnsNormally);
+        final value = doc.parseAt([]).value;
+        expect(value, isA<String>());
+        expect(value, equals(string));
+      },
+          zoneSpecification: ZoneSpecification(
+              print: (Zone self, ZoneDelegate parent, Zone zone,
```
Comment (jonasfj) [inline yaml_edit/test/naughty_test.dart:33]: Can we leave a comment here, that we are doing this to throw away warnings from `package:yaml`.

File: yaml_edit/test/random_test.dart [hunk 2]
```diff
@@ -54,8 +55,13 @@ dev_dependencies:
 ''');
 
       for (var j = 0; j < modificationsPerRound; j++) {
-        expect(
-            () => generator.performNextModification(editor), returnsNormally);
+        runZoned(() {
+          expect(
+              () => generator.performNextModification(editor), returnsNormally);
+        },
+            zoneSpecification: ZoneSpecification(
```
Comment (jonasfj) [inline yaml_edit/test/random_test.dart:62]: Again, let's add a comment that we're doing this hide warnings from `package:yaml`.

Comment (jonasfj) [reply yaml_edit/test/random_test.dart:62]: Also, can we run a single check that this won't eat errors / exceptions? (just a manual check will do, hehe)

From reading: https://api.dart.dev/stable/2.9.0/dart-async/runZoned.html
I don't think this is an error-zone, and only error-zones prevents errors from propagating to the calling context.

**Comment by walnutdust** [comment]
Added comments in both suggested places, and tried adding both `throw ArgumentError.value(2);` and `expect(2, 3)` into the zone and they produced the expected output from `package:test` (i.e. errors/exceptions didn't seem to be eaten)!

**Comment by jonasfj** [comment]
Let's merge when travis passes :D

**Pull request merged**